pkg/applier/applier.go

```diff
@@ -495,7 +493,7 @@ func handleMetrics(ctx context.Context, operation string, err error) {
 // checkInventoryObjectSize checks the inventory object size limit.
 // If it is close to the size limit 1M, log a warning.
 func (s *supervisor) checkInventoryObjectSize(ctx context.Context, c client.Client) {
-	u := newInventoryUnstructured(s.syncKind, s.syncName, s.syncNamespace, s.clientSet.StatusMode)
+	u := newInventoryUnstructured(s.syncName, s.syncNamespace)
```

Comment:
Would it be possible to also move this inventory size checking to the converter?

Reply:
I guess this gets called before Applier.Run... so maybe doing it later isn't great...

Reply:
That seems like a good suggestion - technically we could catch inventory size issues earlier that way (during the InvAddTask before the applier performs actuation).

That being said this seems like something we can treat as a separate enhancement and can address in a separate PR

Reply:
Agreed. Please follow up in a separate PR.